tests/shell/builtin.exitcode.test

```diff
@@ -0,0 +1,74 @@
+# Check if builtin commands properly set their exit codes.
+
+high_exit() {
+    return 42
+}
+
+# break, continue, exit, return and newgrp are not tested.
+COMMANDS=": shift unset export readonly local times eval source exec
+set trap true false pwd echo cd hash type command umask alias unalias
+wait jobs read test [ printf kill getopts fg bg help history fc
```

Comment:
`fc` probably shouldn't be in this list, as it only really has meaning in an interactive shell. (See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/fc.html) It's harmless to have it for now, though.